### PR TITLE
Add support for `"sig"` and `"nav"` to index `axes_manager`

### PR DESCRIPTION
**Pull request by ericpre**
This is a simple change but very convenient!

### Progress of the PR
- [x] Add support for `"sig"` and `"nav"` to index `axes_manager`,
- [x] update docstring (if appropriate),
- [ ] update user guide (I may have another to see where best to update),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
import hyperspy.api as hs
import numpy as np
s = hs.signals.Signal1D(np.ones((10, 10, 10)))

assert s.axes_manager["nav"] == s.axes_manager.navigation_axes
assert s.axes_manager["sig"] == s.axes_manager.signal_axes

# Sum over navigation axis
s_sum_nav = s.sum(axis="nav")

# Sum over signal axis
s_sum_sig = s.sum(axis="sig")
```

### CODE REVIEW

**Comment by ericpre** [comment]
The failure on CI is related to sympy 1.13.0rc1 - see https://github.com/sympy/sympy/issues/26663 and potential fix in https://github.com/sympy/sympy/pull/26678.

**Comment by bjodah** [comment]
@ericpre, I saw the mention in the PR over at SymPy: I'm curious, do you know what `Derivative` instances are present in the expressions you use `lambdify` on?  If so, we should try to amend my PR against the sympy repo with some relevant test cases. In particular, I'd like to add test(s) to `test_lambdify.py` analogous to this one:
https://github.com/sympy/sympy/pull/26678/files#diff-a9bc959f22b154b22a706c75d4927a5a2806ce36f499b8e5b6f993470aebafa3

**Comment by ericpre** [comment]
Thanks @bjodah, I will try to make a minimum example representation of the failure here in the coming days. From a quick look at the log, it most likely have to do with the computation of derivative of these expression:
https://github.com/hyperspy/hyperspy/blob/95c194c3b664f63577408b70e488f2f423177952/hyperspy/_components/voigt.py#L97-L99
and 
https://github.com/hyperspy/hyperspy/blob/95c194c3b664f63577408b70e488f2f423177952/hyperspy/_components/heaviside.py#L61

**Comment by ericpre** [comment]
xref #814.

**Comment by ericpre** [comment]
The test failure with sympy 1.13.0rc1 is fixed in #3388.

**Comment by CSSFrancis** [comment]
@ericpre I can look at this on Tuesday! I'm very excited for this one :) 

+1 to following this up with  something like what had been previously proposed. 

```python
s.axes_manager["nav"].scale=.2

s.axes_manager["nav"](scale=0.2, units="nm", offsets=0.2)
```

The second one looks kind of weird, although still correct and you could do `s.axes_manager.signal_axes(...)`

On another note there are so many more ways to index axes than I previously realized. It would be nice to have an example of how to use things like the imaginary number indexing indexing with the axis name etc in order to set things like the scale. 

Maybe just a basic example showing how to sum data along different axes.

**Comment by ericpre** [comment]
See https://github.com/hyperspy/hyperspy/pull/2756 for the setter.

File: hyperspy/docstrings/signal.py [hunk 1]
```diff
@@ -20,14 +20,16 @@
...
+            the axis in the Signal's `axes_manager` or the axis name. If ``"sig"`` or
+            ``"nav"``, the signal or navigation axis will be used, respectively."""
 
 MANY_AXIS_PARAMETER = """: :class:`int`, :class:`str`, :class:`~hyperspy.axes.DataAxis` or tuple
             Either one on its own, or many axes in a tuple can be passed. In
             both cases the axes can be passed directly, or specified using the
             index in `axes_manager` or the name of the axis. Any duplicates are
-            removed. If ``None``, the operation is performed over all navigation
-            axes (default)."""
+            removed. If ``"sig"`` or ``"nav"``, the signal or navigation axis
```
Comment (francisco-dlp) [inline hyperspy/docstrings/signal.py]: axis -> axes?

**Review by francisco-dlp — APPROVED** [review]
LGTM!

File: hyperspy/docstrings/signal.py [hunk 2]
```diff
@@ -20,14 +20,16 @@
 
 ONE_AXIS_PARAMETER = """: :class:`int`, :class:`str`, or :class:`~hyperspy.axes.DataAxis`
             The axis can be passed directly, or specified using the index of
-            the axis in the Signal's `axes_manager` or the axis name."""
+            the axis in the Signal's `axes_manager` or the axis name. If ``"sig"`` or
```
Comment (jlaehne) [inline hyperspy/docstrings/signal.py]:
```suggestion
            the axis in the Signal's ``axes_manager`` or the *axis name*. If ``"sig"`` or
```

File: hyperspy/docstrings/signal.py [hunk 3]
```diff
@@ -20,14 +20,16 @@
...
+            ``"nav"``, the signal or navigation axis will be used, respectively."""
 
 MANY_AXIS_PARAMETER = """: :class:`int`, :class:`str`, :class:`~hyperspy.axes.DataAxis` or tuple
             Either one on its own, or many axes in a tuple can be passed. In
             both cases the axes can be passed directly, or specified using the
             index in `axes_manager` or the name of the axis. Any duplicates are
```
Comment (jlaehne) [inline hyperspy/docstrings/signal.py]:
```suggestion
            Either a single or multiple axes in a tuple can be passed. In
            both cases, the axes can be passed directly, or specified using the
            index in ``axes_manager`` or the *name of the axis*. Any duplicates are
```

File: hyperspy/signal.py [hunk 4]
```diff
@@ -4613,14 +4613,23 @@ def derivative(self, axis, order=1, out=None, **kwargs):
         """
         # rechunk was a valid keyword up to HyperSpy 1.6
         if "rechunk" in kwargs:
+            # We miss the deprecation cycle for 2.0
+            warnings.warn(
+                "The `rechunk` parameter is not used since HyperSpy 1.7 and"
```
Comment (jlaehne) [inline hyperspy/signal.py:4619]:
```suggestion
                "The ``rechunk`` parameter is not used since HyperSpy 1.7 and"
```

**Review by jlaehne — APPROVED** [review]

**Comment by jlaehne** [comment]
We should probably add a section on accessing the axes and their properties in the documentation on the [Axes handling](https://hyperspy--3385.org.readthedocs.build/en/3385/user_guide/axes.html) page right before "setting axis properties".

File: hyperspy/signal.py (hunk 4 above)
Comment (ericpre) [reply hyperspy/signal.py:4619]: This isn't rst formatting! 😉 Should we keep it as a single quotation mark?

**Comment by ericpre** [comment]
Yes, the documentation on AxesManager will need an overhaul and integrating with the gallery of example, etc. I was thinking that it would be a good time to do after https://github.com/hyperspy/hyperspy/pull/2756.

**Pull request merged**